### PR TITLE
Build: configure: fix overkill format specifier serving uint64_t check

### PR DESCRIPTION
**Pull request by jnpkrn**
This was a chained mess:

* fixed-width integer types from <stdint.h> are usually accompanied with
  respective format specifiers in <inttypes.h>, which should have been
  a first line of the solution

* relying on the compiler erroring out on the incorrect format specifier
  in a compile test is broken (if nothing else, may fail like that for
  unrelated reasons, which also happened in the case referred below),
  especially since it's trivially solvable in a preprocessing test,
  which avoids inteferring with -D_FORTIFY_SOURCE vs. optimization
  switches -- even more so when they are split between CFLAGS and
  CPPFLAGS, see https://bugs.clusterlabs.org/show_bug.cgi?id=5274

### CODE REVIEW

**Comment by jnpkrn** [comment]
It's relatively hard to discover all possible format specifier breakage points
should the expectations not fit the usual conventions, so it will be better to
to fix other ones as they are discovered.

**Comment by jnpkrn** [comment]
Fix the problem missed because of non-clean rebuilding.

**Comment by jnpkrn** [comment]
Heck, it's more twisted than I thought :-/
One of the pre-existing `U64T` usages was with `sscanf`,

File: daemons/fenced/pacemaker-fenced.c [hunk 1]
```diff
@@ -123,7 +126,8 @@ st_ipc_dispatch(qb_ipcs_connection_t * qbc, void *data, size_t size)
     }
 
     crm_element_value_int(request, F_STONITH_CALLOPTS, &call_options);
-    crm_trace("Flags %u/%u for command %u from %s", flags, call_options, id, crm_client_name(c));
+    crm_trace("Flags %" U32T "/%u for command %" U32T " from %s",
+              flags, call_options, id, crm_client_name(c));
```
Comment (kgaillot) [inline daemons/fenced/pacemaker-fenced.c]: This should be hex, oversight in original

**Comment by jnpkrn** [comment]
It was the opposite overkill, I admit.  Refreshed.

**Comment by kgaillot** [comment]
The ppc64le builds are failing due to printf/scanf issues in arch-specific code:

https://ci.kronosnet.org/job/pacemaker-build-all-voting/444/pacemaker-build-all-voting=rhel74z-power8-ppc64le/console

**Comment by kgaillot** [comment]
Also, why not just use the inttypes.h constants and dispense with our own? I wouldn't consider those part of our public API.

**Comment by kgaillot** [comment]
> The ppc64le builds are failing due to printf/scanf issues in arch-specific code:
> 
> https://ci.kronosnet.org/job/pacemaker-build-all-voting/444/pacemaker-build-all-voting=rhel74z-power8-ppc64le/console

We only need to make sure those two tools compile, we don't need to test them. So if you push a commit and the CI passes ppc64le, we're good.

**Comment by jnpkrn** [comment]
> I wouldn't consider those part of our public API

How you came to the conclusion `include/config.h` in a public API?

It shouldn't actually be, but the utter mess makes everything a subject
for interpretation :-/

**Comment by jnpkrn** [comment]
Good that accidental `%` omission was spotted with CI (as you said, that's an
arch-specific thing that was affected). Should be fixed now.

**Comment by kgaillot** [comment]
Thanks, this was a nice improvement.

However the _FORTIFY_SOURCE-in-CPPFLAGS issue still exists due to the other uses of cc_temp_flags. I'm not sure there is a good generic solution, but perhaps we can at least address that specific case by stripping _FORTIFY_SOURCE from CPPFLAGS in cc_temp_flags (and restoring it in cc_restore_flags).

**Comment by kgaillot** [comment]
>> I wouldn't consider those part of our public API
> 
> How you came to the conclusion include/config.h is a public API?

I'm confused -- perhaps you misunderstood? My suggestion was that we get rid of U64T etc., which we can do because they are not part of the public API, and use PRIu64 etc. directly.

> It shouldn't actually be, but the utter mess makes everything a subject
> for interpretation :-/

Embrace the chaos :) Things don't have to be well-specified to work, and less-than-ideal is not inherently problematic. It is a goal to more clearly document and delineate our public API over time, but in real-world practice, it causes no problems.

**Comment by jnpkrn** [comment]
On 24/08/18 16:32 +0000, Ken Gaillot wrote:
>>> I wouldn't consider those part of our public API
>> 
>> How you came to the conclusion include/config.h is a public API?
> 
> I'm confused -- perhaps you misunderstood? My suggestion was that we
> get rid of U64T etc., which we can do because they are not part of
> the public API, and use PRIu64 etc. directly.

Oh yes, misunderstanding on my side.

I considered that, but exactly because nothing externally is going
ever depend on that, it provides a bottom-most backup for systems
where inttypes.h simply doesn't provide those specifiers for whatever
reason, and is also a convenient way to recheck that mismatched
specifiers are indeed being caught by the compiler (IIRC there's
still at least one place we cannot apply convenient format __attribute__
annotation directly), e.g. by:

```
make distclean && ./autogen.sh\
 && env pcmk_cv_decl_inttypes='PRId32 PRIu32 PRIx32 SCNu64' ./configure\
 && make
```

(note that the first one should have been `PRIu64`)

All without possibly messing with system header files that got
included.  And, the main point, without making a single change
into the codebase.

>> It shouldn't actually be, but the utter mess makes everything a subject
>> for interpretation :-/
> 
> Embrace the chaos :) Things don't have to be well-specified to work,
> and less-than-ideal is not inherently problematic. It is a goal to
> more clearly document and delineate our public API over time, but in
> real-world practice, it causes no problems.

Since nobody heavily relies on the C level interfaces, except in
very rare and contained cases now.  That we know of, at least.

Anyway, `include/config.h` is an internal thing by design, clearly.

-- 
Nazdar,
Poki

**Pull request merged**